research-software-directory/RSD-as-a-service
# 604 rsd admin can add new organisation

# RSD admin can add/delete organisation

Closes #604

Changes proposed in this pull request:
* Organisation section is added to admin pages. RSD admin can add new organisation from ROR or manually. Admin can delete an organisation if there are not references to any software or projects.
* Initial order of the organisations in the admin overview list the organisations that can be deleted first. 
* Admin can search organisations by name, website or ROR id
* Admin can edit RSD organisation. The edit button will open the organisation settings page.
* Keyword section is added to admin pages. RSD admin can add/edit/delete keywords. Only keywords that are not refrenced by software or project can be deleted.

How to test:
* `make start` to rebuild app
* login as rsd admin
* navigate to [admin/organisations ](http://localhost/admin/organisations)
* Add organisation from ROR using search input on the right side (see image)
* Remove organisation using delete button. Note! Only organisation without references to software or projects can be removed
* Edit organisation by using edit button. It will open settings page of selected organisation.
* Navigate to keyword administration section
* Add new keyword by typing it in the search bar, then use Add button on the right side
* Edit keyword by clicking on the keyword value (see image). All keywords can be edited.
* Delete an keyword. Only the keywords that are not referenced by any software or project have delete button enabled. Hint: new keyword you just added will be enabled for deletion.

## Admin organisations section
![image](https://user-images.githubusercontent.com/9204081/224777931-9264c8f2-5033-4e26-ad4e-f2bd10999cc9.png)

## Remove organisation
![image](https://user-images.githubusercontent.com/9204081/224782017-3574a205-0224-4a76-9051-94bc11cdac35.png)

## Add new keyword
![image](https://user-images.githubusercontent.com/9204081/224779252-4d640a2c-567d-4441-a5e1-1708a48a7df2.png)

## Edit keyword

![image](https://user-images.githubusercontent.com/9204081/224779640-9acdec2d-4930-4efe-9049-a269585ef9b9.png)


PR Checklist:
